compiler/rustc_span/src/span_encoding.rs

```diff
@@ -102,6 +108,8 @@ impl Span {
 
         let (base, len, ctxt2) = (lo.0, hi.0 - lo.0, ctxt.as_u32());
 
+        // FIXME: THIS IS UNSOUND. WE DO NOT CHECK FOR THE SIZE BEFORE CREATING THE NonMaxU32!!!
+
         if len <= MAX_LEN && ctxt2 <= MAX_CTXT {
```

Comment:
I think this is unsound, we don't check for the size before creating the nonmaxu32